src/gzip/dynamic_compression_handler.cr

```diff
@@ -0,0 +1,57 @@
+{% if !flag?(:without_zlib) %}
+  require "gzip"
+{% end %}
+
+module Lucky::DynamicCompressionHandler
+  COMPRESS_ONLY_THESE_EXTENSIONS = ["", ".html", ".htm", ".txt", ".json", ".xml"]
```

Comment:
This could be a habitat setting as well, that way people can configure it if desired

Reply:
I have to think about that. I'm not sure the size of the footgun. Mostly I'm thinking about binary that shouldn't be compressed. I'm wondering if it should be the framework's responsibility to know what is reasonable to compress? I'm a bit on the fence.

Reply:
will do